### PR TITLE
80 add binary to wrapper function for unanchored case

### PR DESCRIPTION
**Pull request by hoppanda**

### CODE REVIEW

File: tests/testthat/test-maic_unanchored.R [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+      trt_var_agd = "ARM",
+      endpoint_type = "binary",
+      endpoint_name = "Binary Endpoint",
+      eff_measure = "RR",
+      # binary specific args
+      binary_robust_cov_type = "CR2"
+    )
+
+  # save(list = c("expectout","expectout2"), file = file.path("inst","extdata", "test_binary_unanchored_expected.RData"))
+  load(system.file("extdata", "test_binary_unanchored_expected.RData", package = "maicplus", mustWork = TRUE))
```
Comment (gravesti) [inline tests/testthat/test-maic_unanchored.R:63]: we can put this in the test dir, eg find it using `testthat::test_dir()`

File: R/maic_unanchored.R [hunk 2]
```diff
@@ -131,107 +147,269 @@ maic_unanchored <- function(weights_object,
...
+  )
+
+  # : fit glm for binary outcome and robust estimate with weights
+  binobj_dat <- glm(RESPONSE ~ ARM, dat, family = glm_link)
+  binobj_dat_adj <- glm(RESPONSE ~ ARM, dat, weights = weights, family = glm_link)
+  bin_robust_cov <- clubSandwich::vcovCR(binobj_dat_adj,
+    cluster = dat$USUBJID,
+    type = binary_robust_cov_type
+  )
+  bin_robust_coef <- clubSandwich::conf_int(binobj_dat_adj, bin_robust_cov, coef = 2, p_values = TRUE)
```
Comment (MikeJSeo) [inline R/maic_unanchored.R:321]: I think we should use sandwich package and not clubsandwich
let's discuss

**Review by MikeJSeo — CHANGES_REQUESTED** [review]

**Comment by MikeJSeo** [comment]
image
why is some comments hidden?

File: man/get_pseudo_ipd_binary.Rd [hunk 3]
```diff
@@ -0,0 +1,47 @@
+% Generated by roxygen2: do not edit by hand
+% Please edit documentation in R/binary-helper.R
+\name{get_pseudo_ipd_binary}
+\alias{get_pseudo_ipd_binary}
+\title{Helper funciton: create pseudo IPD given AgD binary}
```
[the remark this answers is not in the document]
Comment (hoppanda) [reply man/get_pseudo_ipd_binary.Rd:5]: addressed

File: tests/testthat/test-maic_unanchored.R (hunk 1 above)
Comment (gravesti) [reply tests/testthat/test-maic_unanchored.R:63]: @hoppanda I have this prepared to push. Let me know when you have pushed your latest changes here and I will merge them

File: R/maic_unanchored.R (hunk 2 above)
Comment (gravesti) [reply R/maic_unanchored.R:326]: @MikeJSeo Which sandwich `vcov*()` could we use?

File: R/maic_unanchored.R [hunk 4]
```diff
@@ -131,107 +147,269 @@ maic_unanchored <- function(weights_object,
...
-        format(round(boot_res_AB$ci_u, 2), nsmall = 2), "]"
-      )
-      res$inferential[["report_overall_bootCI"]] <- rbind(
-        report_table(coxobj_dat, medSurv_dat, tag = paste0("Before matching/", endpoint_name)),
-        tmp_report_table
-      )
     }
+    tmp_report_table_binary <- report_table_binary(binobj_dat_adj, res_AB, tag = paste0("After matching/", endpoint_name), eff_measure = eff_measure)
+    tmp_report_table_binary[[paste0(eff_measure, "[95% CI]")]][1] <- paste0(
+      format(round(boot_res_AB$est, 2), nsmall = 2), "[",
```
[the remark this answers is not in the document]
Comment (gravesti) [reply R/maic_unanchored.R]: Made new issue for this #101

File: R/maic_unanchored.R [hunk 5]
```diff
@@ -131,107 +147,269 @@ maic_unanchored <- function(weights_object,
...
+    stop("Endpoint type ", endpoint_type, " currently unsupported.")
+  }
+
+  # output
+  result
+}
+
+# MAIC inference functions for TTE outcome type ------------
+
+maic_unanchored_tte <- function(res,
```
[the remark this answers is not in the document]
Comment (gravesti) [reply R/maic_unanchored.R:188]: New issue #102

File: R/maic_unanchored.R [hunk 6]
```diff
@@ -88,6 +110,7 @@ maic_unanchored <- function(weights_object,
 
     if (any(!c("USUBJID", "RESPONSE") %in% names(ipd))) stop("ipd should have 'USUBJID', 'RESPONSE' columns at minimum")
     eff_measure <- match.arg(eff_measure, choices = c("OR", "RD", "RR"), several.ok = FALSE)
+    # !! add a check of binary_robust_cov_type
```
[the remark this answers is not in the document]
Comment (gravesti) [reply R/maic_unanchored.R]: added check

File: R/maic_unanchored.R (hunk 2 above)
Comment (gravesti) [reply R/maic_unanchored.R]: switched to sandwich wtih HC3 by default. Using lmtest to get CIs

**Review by gravesti — APPROVED** [review]

**Review by MikeJSeo — APPROVED** [review]

**Pull request merged**